test/acceptance/rule-test.js

```diff
@@ -95,6 +95,53 @@ describe('rule public api', function () {
     });
   });
 
+  describe('async rule visitor support', function () {
+    generateRuleTests({
+      plugins: [
+        {
+          name: 'test',
+          rules: {
+            'no-promise': class extends Rule {
+              async visitor() {
+                await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
Can we dial this timeout down? I'd rather the test didn't take 1s (literally any async should have failed before)...

```suggestion
                await new Promise((resolve) => setTimeout(resolve, 10));
```